src/Adapter/AdapterAbstract.php

```diff
@@ -67,16 +65,15 @@ public function ok($output)
      *
      * This assumes the output is in Test Anywhere Protocol (TAP) format.
      *
-     * TODO: Update regex to make certain we take the LAST numbered ok
-     *
      * @param string $output
      * @return bool|int
      */
     public function hasOks($output)
     {
-        $result = preg_match_all("%ok (\\d+) - .*%m", $output, $matches);
+        $result = preg_match_all("%##teamcity\[testFinished%", $output);
         if ($result) {
-            return (int) end($matches[1]);
+            $this->okCount += $result;
+            return $this->okCount; // was: (int) end($matches[1]);
```

Comment:
It probably should, and it's a public method to boot, but something we can clean up with greater forethought for 2.0 when defining interfaces/extension points.